distribution/src/bin/opensearch-env

```diff
@@ -78,7 +78,9 @@ fi
 
 export HOSTNAME=$HOSTNAME
 
-${source.path.env}
+if [ -z "$OPENSEARCH_PATH_CONF" ]; then
+  ${source.path.env}
```

Comment:
Currently this sources this [file](https://github.com/opensearch-project/OpenSearch/blob/main/distribution/packages/src/common/env/opensearch).  This has other environment variables which won't be set if there is an environment variable `OPENSEARCH_PATH_CONF` in the current shell. 
This might create problems without the user realizing why.

I think this needs be properly communicated in documentation and by logging a warning message.

Reply:
I agree there needs a properly notice to the users to aware the other environment variable in `${source.path.env}`, because there is one:
https://github.com/opensearch-project/OpenSearch/blob/34550c5b17124ddc59458ef774f6b43a086522e3/distribution/packages/src/common/env/opensearch#L31
But during my test, I found the variables in above file can still be set through systemd service file: https://github.com/opensearch-project/OpenSearch/blob/34550c5b17124ddc59458ef774f6b43a086522e3/distribution/packages/src/common/systemd/opensearch.service#L15 . After the users commenting out this line manually, they can override the value of `OPENSEARCH_PATH_CONF`, and start several OpenSearch processes in the same host.
So in my opinion, the users understand what they are doing because they still need to do the above additional step, and common users will not be affected by the change in the PR.

Reply:
That's (`OPENSEARCH_STARTUP_SLEEP_TIME=5 `) the only uncommented one but there are other variables that might have modified. The service can be started without `systemd`.  We can assume that the users know what they are doing but if we're changing existing behaviour it should be properly communicated.

Reply:
Thanks Rabi 😄 I see, you are right.
With the change in the PR, when user set custom `OPENSEARCH_PATH_CONF`, and started OpenSearch process without systemd (such as running the `opensearch` bash script directly) the default values in `${source.path.env}` will not be loaded, so it needs a notice.